[Code Style] Enforce code style

VWoeltjen opened the pull request:
Enforce code style using `jscs` as part of the `gulp verify` task; add rules for code style; modify existing code to comply. Addresses #142.

This is probably easier to review by commit:

* 031a46aa8e34f2e428c71b9cf9e4c83412c8d438 contains changes to JSCS rules and updates to the Gulp build.
* fa77139077256b4d55669b1e3d2ae779d60b6a71 updates sources using `gulp fixstyle` (uses `jscs` to normalize code style)
* f9631ff4c5e37bb6485d45b4c852b2505ec582ff fixes the one remaining code style violation not corrected by the above.

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y (updated to match code style settings!)
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

akhenry commented:
Changes look good otherwise. Only reviewed a sample of the gulp fixstyle changes, but they looked fine.

I'll leave it up to you whether to use outer or inner. If it's going to cause more pain to fix the codebase than it saves us in potential bugs down the track, then sticking with inner is fine. 

If you change it, and the code diff is substantial, I'm happy to have another look at the PR.

Also, looks like recent merges into master have caused conflicts. Apologies, it's my fault for sitting on this for a week. Might be easiest to rebase -i over master, deleting fa77139, and then just re-run gulp fixstyle?

### Author Checklist

1. Changes address original issue?
2. Unit tests included and/or updated with changes?
3. Command line build passes?
4. Changes have been smoke-tested?

VWoeltjen commented:
Changed `shadow` to `outer`, renamed variables which violated this rule, merged in latest from master, ran fixstyle on the result.

@akhenry - think this probably does deserve another review, in particular the variables renamed in ad5691142e076b52c127211c037f20f2759d5886 and the conflict resolution in bce56439942a8f2d355b565873df7aa7ee29c1f7

akhenry commented:
Looks good to me. Had a look through the renamed variables and couldn't see any problems.

### Author Checklist

1. Changes address original issue? __Y__
2. Unit tests included and/or updated with changes? __Y__
3. Command line build passes? __Y__
4. Changes have been smoke-tested? __Y__
